### PR TITLE
feat: update to Angular 16

### LINKED ISSUE #22: update to Angular 16
**Describe the feature:**
Update the library to the latest version of Angular (version 16), including support for signals, standalone components, and any other new features or adjustments as appropriate.

### PR DESCRIPTION
**Pull request by samrichardsontylertech**
## PR Checklist

Please check if your PR fulfills the following requirements:

- Tests for the changes have been added: N/A
- Docs have been added / updated: N/A
- Does this PR introduce a breaking change? Y
- I have linked any related GitHub issues to be closed when this PR is merged? Y

## Describe the new behavior?
Resolves #22 

The adapter library has been updated to Angular 16. Proxy component modules are now re-exported from the base component modules and the bucket proxy component module has been removed as `ForgeModule` now includes everything.

The demo app has also been updated with stronger typing, typed reactive forms, and component icons to reflect proper usage.

## Additional information
Further work will need to be done to begin transitioning dynamic components away from component factories before Angular removes them from the API.

### CODE REVIEW

File: src/app/views/components/button/button.component.html [hunk 1]
```diff
@@ -22,14 +22,14 @@
...
+        <forge-icon name="favorite"></forge-icon>
         <span>Leading icon</span>
       </button>
     </forge-button>
     <forge-button type="outlined">
       <button>
         <span>Trailing icon</span>
-        <i class="tyler-icons">favorite</i>
+        <forge-icon name="favorite"></forge-icon>
       </button>
```
Comment (DRiFTy17) [inline src/app/views/components/button/button.component.html:33]: More instances of `type="button"`

File: src/app/views/components/bottom-sheet/lazy/lazy-bottom-sheet.component.html [hunk 2]
```diff
@@ -6,9 +6,9 @@ <h5 class="forge-dialog__title">{{title}}</h5>
...
-    <button (click)="onCancel()">CANCEL</button>
+    <button (click)="onCancel()">Cancel</button>
   </forge-button>
   <forge-button>
-    <button forge-bottom-sheet-focus (click)="onConfirm()">CONFIRM</button>
+    <button forge-bottom-sheet-focus (click)="onConfirm()">Confirm</button>
   </forge-button>
-</footer>
\ No newline at end of file
+</footer>
```
Comment (DRiFTy17) [inline src/app/views/components/bottom-sheet/lazy/lazy-bottom-sheet.component.html]: More `type="button"`

File: src/app/views/components/bottom-sheet/confirm/confirm-bottom-sheet.component.html [hunk 3]
```diff
@@ -6,9 +6,9 @@ <h5 class="forge-dialog__title">{{title}}</h5>
 </section>
 <footer class="forge-dialog__footer">
   <forge-button>
-    <button (click)="onCancel()">CANCEL</button>
+    <button (click)="onCancel()">Cancel</button>
   </forge-button>
   <forge-button>
-    <button forge-bottom-sheet-focus (click)="onConfirm()">CONFIRM</button>
+    <button forge-bottom-sheet-focus (click)="onConfirm()">Confirm</button>
   </forge-button>
```
Comment (DRiFTy17) [inline src/app/views/components/bottom-sheet/confirm/confirm-bottom-sheet.component.html:13]: More `type="button"`

File: src/app/views/examples/table-example/table-example.component.ts [hunk 4]
```diff
@@ -114,9 +115,8 @@ export class TableExampleComponent implements OnInit {
         value: index,
         leadingBuilder: () => {
           if (isSelected) {
-            const icon = document.createElement('i');
-            icon.classList.add('tyler-icons');
-            icon.textContent = 'check';
+            const icon = document.createElement('forge-icon') as IconComponent;
+            icon.name = 'check';
             return icon;
```
[the remark this answers is not in the document]
Comment (samrichardsontylertech) [reply src/app/views/examples/table-example/table-example.component.ts]: I switched over most of the code like this to use component delegates. It's more concise in a few place but I noticed a couple areas that could use some streamlining (adding click listeners to buttons and `aria-label`s where needed without going into the structure of the elements).

**Review by DRiFTy17 — APPROVED** [review]

**Pull request merged**